### ISSUE TITLE
Questions and parameter optimization running scAR on CITEseq data

### ISSUE THREAD

**Issue by LucHendriks** [Novartis/scAR#93]
Thank you for developing and maintaining this package, I have been working with it for a little bit now and have a couple questions regarding how to optimize certain parameters.

### `setup_anndata()`
- `prob`: I am running scAR for CITEseq data and I followed the tutorial on how to denoise ADT counts. In this tutorial a prob value of 0.7 is used and it is mentioned that you should use a lower prob value if you get errors. Am I right in interpreting this as use a as high as possible prob value unless you get errors? In other words, should I use the default 0.995 if it does not give me errors? 
- `iterations`: Should I play around with the parameter or best left at default?
- kneeplot: When verifying the empty droplet selection in the kneeplot, is there a specific part of the kneeplot that should be empty droplets to best work with scAR? I want to use this to verify that my prob value led to the right selection as well.

### `model()`
- NN: is there a different default config (nn_layer1, nn_layer2, latent_dim) that should be used for the NN for different feature types, in specific for me I am interested if I should adjust this for CITEseq data.
- `count_model`: Could you maybe give me a bit more info on when to use a certain count_model, or maybe some basic guidelines on this?
- `model.train()`:
  - `epochs`: Is there any way to optimize the number of epochs that needs to be used?
  - Is there any other parameters I should tweak, or is it better to leave it on the default values?
- `model.inference()`:
  - `count_model_inf`: Should this be the same value as the `count_model` in the model setup?
  - `get_native_frequencies`: Should this be something I should consider to use for CITEseq data?

Thank you in advance for the help!

**Comment by CaibinSh** [comment]
Hi @LucHendriks , 

Thank you again for the interest. We optimized the model on multiple datasets, so I would recommend using the default settings where possible.

> Thank you for developing and maintaining this package, I have been working with it for a little bit now and have a couple questions regarding how to optimize certain parameters.
> 
> ### `setup_anndata()`
> * `prob`: I am running scAR for CITEseq data and I followed the tutorial on how to denoise ADT counts. In this tutorial a prob value of 0.7 is used and it is mentioned that you should use a lower prob value if you get errors. Am I right in interpreting this as use a as high as possible prob value unless you get errors? In other words, should I use the default 0.995 if it does not give me errors?

In some cases, a high `prob` value results in zero cell-free droplets being identified, which will cause an error. Therefore, we suggested lowering the prob value cutoff to address these cases in the tutorial.

We recommend using the knee plot to determine the cutoff. Based on EmptyDrops, droplets with a significantly greater than 0 prob value in the knee plot are considered `cell-free droplets`. You can identify the maximal number of cell-free droplets by tuning the prob parameter in `setup_anndata()`. The lower `prob` in parameter, the more cell-free droplets called, the more accurate estimation for the ambient profile. On the other hand , too low `prob` has a risk to falsely call `other droplets` (e.g., droplets containing cell debris) as `cell-free droplets`.  Knee plot can tell us where the cutoff might be.

A higher prob value leads to more stringent criteria for identifying cell-free droplets, resulting in fewer droplets being classified as cell-free, this might also not be a good idea.

> * `iterations`: Should I play around with the parameter or best left at default?
Three iterations are sufficient.

> * kneeplot: When verifying the empty droplet selection in the kneeplot, is there a specific part of the kneeplot that should be empty droplets to best work with scAR? I want to use this to verify that my prob value led to the right selection as well.

The so-called "cell-free droplets" are the population that matters. We use only this population to calculate the ambient profile.


> ### `model()`
> * NN: is there a different default config (nn_layer1, nn_layer2, latent_dim) that should be used for the NN for different feature types, in specific for me I am interested if I should adjust this for CITEseq data.

We have optimized these parameters, and the default settings should also work well for CITEseq data.

> * `count_model`: Could you maybe give me a bit more info on when to use a certain count_model, or maybe some basic guidelines on this?

Please use the default 'binomial' model. We benchmarked several models, but the default one consistently gave the best results.

> * `model.train()`:
>   * `epochs`: Is there any way to optimize the number of epochs that needs to be used?

300-400 epochs are sufficient. More epochs do not lead to significantly different results. Depending on your GPU power, you can train for additional epochs if desired, but it’s not necessary.

>   * Is there any other parameters I should tweak, or is it better to leave it on the default values?

I would recommend leaving them at the default values if the results are satisfactory for you.

> * `model.inference()`:
 
>   * `count_model_inf`: Should this be the same value as the `count_model` in the model setup?

Not exactly; there is a slight difference. I recommend leaving it at the default, i.e., poisson.


>   * `get_native_frequencies`: Should this be something I should consider to use for CITEseq data

`get_native_frequencies` determines whether to output normalized expression levels per cell (summing to one). It does not influence model training or inference. We set it to False by default to save memory.

> Thank you in advance for the help!

You are welcome. Please let me know if there are other questions.
